.github/workflows/MainDistributionPipeline.yml

```diff
@@ -0,0 +1,29 @@
+name: Main Extension Distribution Pipeline
+on: [push, pull_request, repository_dispatch]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref || '' }}-${{ github.base_ref || '' }}-${{ github.ref != 'refs/heads/main' || github.sha }}
+  cancel-in-progress: true
+
+# TODO: move workflows to official repo
+jobs:
+  linux-distribution:
+    name: Linux
+    uses: samansmink/extension-workflows/.github/workflows/Linux.yml@main
+    with:
+      extension_name: quack
```

Comment:
Minor, can the extension name be a top-level environment variable, then passed over the workflows?

Also it would be nice, but unsure how to do this exactly, to have first all test performed and all stuff uploaded, then a separate (optional) deploy step (so you can say, you deploy only if all success).

Reply:
Basic would be having Deploy.yml be a separate wofklow, taking as parameters the relevant things like platform & co.

Reply:
> Minor, can the extension name be a top-level environment variable, then passed over the workflows?

Well the surprising answer is no, at least not in a clean way: https://stackoverflow.com/questions/73797254/environment-variables-in-github-actions